src/components/Checkbox.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+  <label :for="id" class="checkbox-label" :class="labelClasses">
+    <input
+      :id="id"
+      type="checkbox"
+      class="checkbox"
+      :checked="checked"
+      :name="inputName"
+      :value="inputValue"
+      :disabled="disabled"
+      @change="onChange"
+    /><span>
+      <slot>
+        {{ label }}
+      </slot>
+    </span>
+  </label>
+</template>
+
+<script>
+export default {
+  name: 'Checkbox',
```

Comment:
Single word component names are not recommended in Vue because you could inadvertently override some future HTML native element by the same name (all native elements have a single word name, even `textarea` which is two words in the prototype name `HTMLTextAreaElement` 😱)

For that reason I was appending `V` to `VButton` for example. This helps future proof the Vue components (even though it makes the names less pleasant 😞)